src/test/java/ay2122s1_cs2103t_w16_2/btbb/logic/commands/client/FindClientCommandTest.java

```diff
@@ -80,31 +78,31 @@ public void equals() {
     @Test
     public void execute_multipleKeywords_multipleClientsFound() {
         String expectedMessage = String.format(MESSAGE_CLIENTS_LISTED_OVERVIEW, 3);
-        ClientPredicateCollection clientPredicateCollection = new ClientPredicateCollection();
-        clientPredicateCollection.addClientPredicate(
-                prepareAndGetClientPredicate("Kurz Elle Kunz", NameContainsKeywordsPredicate::new)
+        PredicateCollection<Client> predicateCollection = new PredicateCollection<>();
+        predicateCollection.addPredicate(
+                prepareAndGetClientPredicate("Kurz Elle Kunz", Client::getName)
         );
-        clientPredicateCollection.addClientPredicate(
-                prepareAndGetClientPredicate("9535 9482 2427", PhoneContainsKeywordsPredicate::new)
+        predicateCollection.addPredicate(
+                prepareAndGetClientPredicate("9535 9482 2427", Client::getPhone)
         );
-        clientPredicateCollection.addClientPredicate(
-                prepareAndGetClientPredicate("wall michegan tokyo", AddressContainsKeywordsPredicate::new)
+        predicateCollection.addPredicate(
+                prepareAndGetClientPredicate("wall michegan tokyo", Client::getAddress)
         );
-        clientPredicateCollection.addClientPredicate(
-                prepareAndGetClientPredicate("heinz werner lydia", EmailContainsKeywordsPredicate::new)
+        predicateCollection.addPredicate(
+                prepareAndGetClientPredicate("heinz werner lydia", Client::getEmail)
         );
-        FindClientCommand command = new FindClientCommand(clientPredicateCollection);
-        expectedModel.updateFilteredClientList(clientPredicateCollection);
+        FindClientCommand command = new FindClientCommand(predicateCollection);
+        expectedModel.updateFilteredClientList(predicateCollection);
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(Arrays.asList(CARL, ELLE, FIONA), model.getFilteredClientList());
     }
 
     /**
      * Parses {@code userInput} into a {@code Predicate<Client>}.
      */
-    private Predicate<Client> prepareAndGetClientPredicate(String input,
-                                                         Function<List<String>, Predicate<Client>> predicateFunction) {
+    private StringContainsKeywordPredicate<Client> prepareAndGetClientPredicate(String input,
```

Comment:
Can we put this in `testutil.PredicateUtil`? I'm also not so sure for now. But the reason is that this is a util method to make predicates, it is likely to be used in different `FindXCommandTest`. Also we cannot assume that `ClientPredicate == StringContainsKeywordPredicate`
```suggestion
    private <T> StringContainsKeywordPredicate<T> makeStringContainsKeywordPredicate(String input,
```